src/Core/Router.php

```diff
@@ -2,60 +2,56 @@
 
 namespace TrendingRepos\Core;
 
+use TrendingRepos\Exceptions\RouterException;
+
 class Router {
-    public $routes = [
-        'GET' => [],
-        'POST' => []
-    ];
 
-    public static function load($file) {
-        $router = new static;
-        require $file;
+    public $routes;
+    public $request;
 
-        return $router;
+    public function __construct(array $routes, $request)
+    {
+        $this->routes = $routes;
+        $this->request = $request;
+        $this->get();
     }
 
-    public function get($uri, $controller) {
-        $parameters = (new Request)->parameters();
-        if(!empty($parameters)) {
-            foreach($parameters as $key => $value) {
-                $uri = trim($uri, $key . '=' . $value);
+    public function get() {
+        try {
+            if($this->uriExists()){
+                $this->direct($this->request->getSlug(), $this->request->methodType());
+            }else {
+                throw new RouterException('This is not the web page you are looking for!');
             }
-            $uri = trim($uri, '/?');
-
-           if($uri  == null) {
-               $uri = '';
-           }
         }
-
-        $this->routes['GET'][$uri] = $controller;
-    }
-
-    public function post($uri, $controller) {
-        $this->routes['POST'][$uri] = $controller;
+        catch(RouterException $e) {
+            echo $e->getErrorMsg();
+        }
     }
 
-    public function direct($uri, $methodType) {
-
-        if(array_key_exists($uri, $this->routes[$methodType])) {
-            $some = explode('@',$this->routes[$methodType][$uri]);
+    public function direct(string $uri, string $methodType) {
+        try {
 
             return $this->callAction(
                 ...explode('@', $this->routes[$methodType][$uri])
             );
-//            return $this->routes[$methodType][$uri];
+        }catch(RouterException $e) {
+            echo $e->getErrorMsg(); 
         }
-        return view('404');
     }
 
-    public function callAction($controller, $action){
+    public function callAction(string $controller, string $action){
         $controller = "TrendingRepos\\Controller\\{$controller}";
         $controller = new $controller;
 
         if(! method_exists($controller, $action)){
```

Comment:
Space is not needed after negation but needed before opening curly brace:
```suggestion
        if(!method_exists($controller, $action)) {
```